crates/ark/src/data_explorer/table.rs

```diff
@@ -0,0 +1,61 @@
+//
+// table.rs
+//
+// Copyright (C) 2024 by Posit Software, PBC
+//
+//
+
+use anyhow::anyhow;
+use dashmap::DashMap;
+use harp::RObject;
+use once_cell::sync::Lazy;
+
+use crate::thread::RThreadSafe;
+
+// Stores the table R objects objects that contain the data for each
+// data explorer instance.
+// This allows for background threads to easilly access the
+// instance related data without having to rely on the lifetime
+// of data explorer execution thread.
+// Since this is a DashMap, it's safe to access it's underlying data from
```

Comment:
Instead of a dashmap stored in a global, could we use an `Arc<Mutex<HashMap>>`? This should solve the lifetime concerns?

Reply:
Or passing around `Arc<Mutex<Option<RThreadSafeObject<RObject>>>>` as we discussed on Slack, so that an instance can set it to `None` when it's dropped, and getting a `None` would be a cancellation point for the task.

Reply:
This worked nice! So got rid of the global dashmap, and lifetime is now managed with `Arc`. Failing to `get()` from the table means that the table has been deleted and so it can be used to cancel tasks in case the data explorer is closed, although we don't really use it for now.

We would probably also want to implement a cancelling mechanism based on the `callback_id`, so the front-end can cancel tasks it will no longer need. But we can leave this for another PR.

Reply:
oh I meant cancelling the idle task, which does happen already via the propagated `get()` error, e.g. your comment

```rust
        // In case something goes wrong whhile computing the profiles, we send
        // an empty response. Ideally, we would have a way to comunicate an that
        // an error happened but it's not implemented yet.
```